File: Algorithms/src/1574. Shortest Subarray to be Removed to Make Array Sorted/Code/solution.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        # Base case
        if n == 1:
            return "0"

        length = (1 << n) - 1   # 2^n - 1
        mid = (length + 1) // 2

        if k == mid:
            return "1"
        elif k < mid:
            return self.findKthBit(n - 1, k)
        else:
            bit = self.findKthBit(n - 1, length - k + 1)
            return "1" if bit == "0" else "0"

File: Algorithms/src/1574. Shortest Subarray to be Removed to Make Array Sorted/Code/solution.cpp
class Solution {
public:
    char findKthBit(int n, int k) {
        // Base case
        if (n == 1) return '0';

        int length = (1 << n) - 1;        // 2^n - 1
        int mid = (length + 1) / 2;       // middle position

        if (k == mid) {
            return '1';
        } 
        else if (k < mid) {
            // Left half
            return findKthBit(n - 1, k);
        } 
        else {
            // Right half
            char bit = findKthBit(n - 1, length - k + 1);
            // Invert the result
            return bit == '0' ? '1' : '0';
        }
    }
};